staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -442,10 +437,23 @@ func (o *ExposeServiceOptions) createService() (*corev1.Service, error) {
 			case len(protocol) == 0 && len(portProtocolMap) > 0:
 				// no --protocol and we expose a multiprotocol resource
 				protocol = "TCP" // have the default so we can stay sane
-				if exposeProtocol, found := portProtocolMap[stillPortString]; found {
-					protocol = exposeProtocol
+				if exposeProtocols, found := portProtocolMap[stillPortString]; found {
+					for _, exposeProtocol := range exposeProtocols {
+						name := fmt.Sprintf("%s-%d", strings.ToLower(exposeProtocol), port)
+						ports = append(ports, corev1.ServicePort{
+							Name:     name,
+							Port:     int32(port),
+							Protocol: corev1.Protocol(exposeProtocol),
+						})
+					}
+					continue
 				}
 			}
+
+			name := servicePortName
+			if len(portStringSlice) > 1 {
+				name = fmt.Sprintf("%s-%d", strings.ToLower(protocol), port)
+			}
```

Comment:
Compared with the naming of prot-i, this method may be easier to understand, which is also to distinguish between different protocols on the same port